src/display/graphics.cpp

```diff
@@ -400,16 +400,16 @@ struct Movie
                 THEORAPLAY_freeAudio(audioQueueHead->audio);
             }
             audioQueueHead = NULL;
+            SDL_DestroyMutex(audioMutex);
+            audioThreadTermReq.set();
+            alSourceStop(audioSource);
+            alDeleteSources(1, &audioSource);
+            alDeleteBuffers(STREAM_BUFS, alBuffers);
         }
-        SDL_DestroyMutex(audioMutex);
-        audioThreadTermReq.set();
         if(audioThread) {
             SDL_WaitThread(audioThread, 0);
             audioThread = 0;
         }
```

Comment:
Does the re-ordering of these lines relative to the surrounding lines change the behavior at all? (I'm not familiar enough with the audio code to evaluate whether that's an issue.)

Reply:
There was no significance to the order change aside from putting it into a conditional. The variables in question are only populated if there was audio to begin with. For a theora video with no audio, I had crashes before the change but no issues after.

Reply:
OK, sounds good to me then. Everything else looks sane at first glance, though I haven't tested the change here.